rpmlint/checks/PolkitCheck.py

```diff
@@ -0,0 +1,128 @@
+        """Checks files in polkit-default-privs.d."""
+        prefix = '/etc/polkit-default-privs.d/'
+        profiles = ('restrictive', 'standard', 'relaxed')
+        permfiles = []
+        # first pass, find additional files
+        for f in pkg.files:
+            if f.startswith(prefix):
+                if f in pkg.ghostFiles():
+                    self.output.add_info('E', pkg, 'polkit-ghost-file', f)
+                    continue
+
+                bn = f[len(prefix):]
+                if bn not in self.polkit_privs_whitelist:
+                    self.output.add_info('E', pkg, 'polkit-unauthorized-file', f)
+
+                parts = bn.rsplit('.', 1)
+                if len(parts) == 2 and parts[-1] in profiles:
+                    bn = parts[0]
+
+                if bn not in permfiles:
+                    permfiles.append(bn)
+
+        for f in sorted(permfiles):
+            f = pkg.dirName() + prefix + f
+            for profile in profiles:
+                path = '.'.join((f, profile))
+                if os.path.exists(path):
+                    self._parse_privs_file(path)
+                    break
+            else:
+                self._parse_privs_file(f)
+
+    def check_actions(self, pkg):
+        """Checks files in the actions directory."""
+        prefix = '/usr/share/polkit-1/actions/'
+        for f in pkg.files:
+            # catch xml exceptions
+            try:
+                if f.startswith(prefix):
+                    if f in pkg.ghostFiles():
+                        self.output.add_info('E', pkg, 'polkit-ghost-file', f)
+                        continue
+
+                    xml = parse(pkg.dirName() + f)
+                    for a in xml.getElementsByTagName('action'):
+                        self.check_action(pkg, a)
+            except Exception as x:
+                self.output.add_info('E', pkg, 'rpmlint-exception', f'{file:s} raised an exception: {x:s}')
+                continue
+
+    def check_action(self, pkg, action):
+        """Inspect a single polkit action used by an application."""
+        action_id = action.getAttribute('id')
+        if action_id in self.privs:
+            # the action is explicitly whitelisted, nothing else to do
+            return
+
+        allow_types = ('allow_any', 'allow_inactive', 'allow_active')
+        foundunauthorized = False
+        foundno = False
+        foundundef = False
+        settings = {}
+        try:
+            defaults = action.getElementsByTagName('defaults')[0]
+            for i in defaults.childNodes:
+                if not i.nodeType == i.ELEMENT_NODE:
+                    continue
+
+                if i.nodeName in allow_types:
+                    settings[i.nodeName] = i.firstChild.data
+        except KeyError:
+            foundunauthorized = True
+
+        for i in allow_types:
+            if i not in settings:
+                foundundef = True
+                settings[i] = '??'
```

Comment:
This old logic is not quite correct. There are no "undefined" action settings, because there is a default applied in polkit if it is not explicitly stated. This default is not well documented, however from my notes I gather the following:

```
If a setting is missing from the XML policy file then by default `no` is
implied. This is not well documented and needs to be tracked down in the
polkit code, where the `ParserData` is zero initialized and a zero implicit
authorization means `no`.
```

This means that instead of `'??'`, `'no'` should be used here.